dfusion_rust_core/src/models/order.rs

```diff
@@ -32,9 +38,40 @@ impl Serializable for u128 {
     }
 }
 
+impl From<Arc<Log>> for Order {
+    fn from(log: Arc<Log>) -> Self {
+        let mut bytes: Vec<u8> = log.data.0.clone();
+        Order {
+            slot: U256::pop_from_log_data(&mut bytes),
+            slot_index: u16::pop_from_log_data(&mut bytes),
+            account_id: u16::pop_from_log_data(&mut bytes),
+            sell_token: u8::pop_from_log_data(&mut bytes),
+            buy_token: u8::pop_from_log_data(&mut bytes),
+            sell_amount: u128::pop_from_log_data(&mut bytes),
+            buy_amount: u128::pop_from_log_data(&mut bytes),
+        }
+    }
+}
+
+impl From<Entity> for Order {
+    fn from(entity: Entity) -> Self {
+        Order {
+            slot: U256::from_entity(&entity, "slot"),
+            slot_index: u16::from_entity(&entity, "slotIndex"),
+            account_id: u16::from_entity(&entity, "accountId"),
+            sell_token: u8::from_entity(&entity, "sellToken"),
+            buy_token: u8::from_entity(&entity, "buyToken"),
+            sell_amount: u128::from_entity(&entity, "sellAmount"),
+            buy_amount: u128::from_entity(&entity, "buyAmount"),
+        }
+    }
+}
+
 impl From<mongodb::ordered::OrderedDocument> for Order {
     fn from(document: mongodb::ordered::OrderedDocument) -> Self {
         Order {
+            slot: U256::from(document.get_i32("slot").unwrap()),
```

Comment:
Did you check the record inside mongo db? There is no field slot on order. For some reason it's called auctionId there.